From 97b34356fe9919d2ea52f70f2ecb0a94a9b1999a Mon Sep 17 00:00:00 2001
From: nift4 <nift4@protonmail.com>
Date: Thu, 17 Aug 2023 21:36:14 +0200
Subject: [PATCH 2/5] Provide 0 IMEI unless we are emergency-calling

As per 90d6826548189ca850d91692e71fcc1be426f453 ("Remove sensitive info
from SUPL requests"), except in case of emergency, do not include sensitive
subscriber information in SUPL requests (A-GPS).
---
 smali/com/mediatek/location/lppe/main/LPPeService.smali | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/smali/com/mediatek/location/lppe/main/LPPeService.smali b/smali/com/mediatek/location/lppe/main/LPPeService.smali
index 02f8735..a8d4fc8 100644
--- a/smali/com/mediatek/location/lppe/main/LPPeService.smali
+++ b/smali/com/mediatek/location/lppe/main/LPPeService.smali
@@ -3045,9 +3045,14 @@
 .end method
 
 .method private lbsProvideCapabilitiesHandler()V
-    .locals 4
+    .locals 5
+
+    invoke-direct {p0}, Lcom/mediatek/location/lppe/main/LPPeService;->isInEmergency()Z
+
+    move-result v4
 
     .line 1553
+
     iget-object v0, p0, Lcom/mediatek/location/lppe/main/LPPeService;->mTelephonyMgr:Landroid/telephony/TelephonyManager;
 
     const/4 v1, 0x0
@@ -3056,6 +3061,8 @@
 
     const-string v3, ""
 
+    if-eqz v4, :cond_1
+
     if-eqz v0, :cond_1
 
     .line 1554
-- 
2.56.0

